### DISCUSSION TITLE
trame url add prefix

### DISCUSSION THREAD

**Discussion by haoxl3** [Kitware/trame#398] [Q&A]
when I run this command,
`python3 ./pcl_visualization/trame_vis.py --port 8082 --server --host 0.0.0.0`
 I get `http://localhost:8082/index.html`, I want to add a prefix to the url like `http://localhost:8082/heart/index.html`, I try to `python3 ./pcl_visualization/trame_vis.py --port 8082 --server --host 0.0.0.0 --c heart`, but I got faild. How do I implement prefix letters in index.html ?

**Comment by jourdain** [comment]
Why do you need a sub-path when you run your app from the command line? The command line execution is really meant for a single user and usually for development. 

If you aim to achieve a real deployment, you should look into docker and probably rely on some front-end proxy to enable SSL.

**Comment by haoxl3** [comment]
> Why do you need a sub-path when you run your app from the command line? The command line execution is really meant for a single user and usually for development.
> 
> If you aim to achieve a real deployment, you should look into docker and probably rely on some front-end proxy to enable SSL.
Because I need to use the ingress of ECS to forward through the subpath, If there is no subpath, it cannot be forwarded directly

**Reply by SergeyYakubov** [reply]
We experience the same problem. 

We need to make a trame app accessible at a specific path like `https://some_site:443/interactivetools/xxx/index.html`, which then redirects it to `http://some_other_site:8080/interactivetools/xxx/index.html`. Nginx can do that, but only if Trame supports subpaths. 

I tried to redirect to `http://some_other_site:8080/index.html`, but it did not work for me.

**Reply by jourdain** [reply]
Why don't you serve the html/js/css content by nginx? Trame does not need to serve the HTTP content; it is just a convenient part when running locally for development. The only important part is the WebSocket connection.

**Reply by SergeyYakubov** [reply]
trying to make that work. Getting a couple of errors in dev console
```
POST http://localhost:8081/paraview/ 404 (Not Found)

GET http://localhost:8081//__trame_router3/trame-router.umd.js net::ERR_ABORTED 404 (Not Found)

index-fdebfea4.js:38 Uncaught (in promise) Event {isTrusted: true, type: 'error', target: script, currentTarget: null, eventPhase: 0, …}
```

**Reply by jourdain** [reply]
ok? I'm not sure I'm following what you are trying to share here.

**Reply by SergeyYakubov** [reply]
I generated static web content as you recommended using this [utility](https://trame.readthedocs.io/en/latest/tools.www.html). When I try to serve this content, I get the errors I mentioned about. One is about missing `/paraview/` file, one is about missing `trame-router.umd.js `, and one is `Uncaught event`

**Reply by jourdain** [reply]
Ok, and apart from the HTTP requests, do you see the files on the disk after running the utility? Also did you specify which client version the tool should generate? Does it match your application?  

Sidenote, `/paraview/` is the launcher so that could be fine if you already manage the trame process on your side.

**Reply by SergeyYakubov** [reply]
Yes, the files are there (except for router)
```
__trame_components	__trame_vtk		index.html
__trame_plotly		__trame_vuetify3	logo.png
__trame_simput		assets			vue.global.js
```
I specified `--client-vesion=vue3` and my client is vue3.

**Reply by jourdain** [reply]
and trame-router is available from the environment where you run `python -m trame.tools.www`? I just tried and it get exported.

**Reply by SergeyYakubov** [reply]
Ah, of course not. It works now that I use the correct environment.

Thank you!
